real_decoder.go

```diff
@@ -444,29 +444,45 @@ func (rd *realFlexibleDecoder) getBytes() ([]byte, error) {
 	return rd.getRawBytes(length)
 }
 
+func (rd *realFlexibleDecoder) getStringLength() (int, error) {
+	length, err := rd.getUVarint()
+	if err != nil {
+		return 0, err
+	}
+
+	n := int(length - 1)
+
+	switch {
+	case n < -1:
+		return 0, errInvalidStringLength
```

Comment:
🤔 Duplicated testing, though here we’re only testing `-1`, below we test for the stronger `< 0`.

Reply:
Hm… those checks ought to probably be inverted anyways. Returning an invalid array length has value above simply returning insufficient data. 🤔 

And yeah, checking the remaining for an array… 🤔 maybe we could pass in a byte size, allowing us to properly check for insufficient data. But as you say… maybe better to just do the check downstream.